### ISSUE TITLE
execution hangs after a few interactions

### ISSUE THREAD

**Issue by JanRiebe** [nobodywho-ooo/nobodywho#38]
I'm using a very simple setup and the suggested Gemma model.
After around 4 interactions (calling say() until receiving _on_completion_finished signal), the execution of the Godot app hangs completely. 

I appreciate any input how to further debug this. Thank you!

I'm running this on an M3 MacBook Air.
Godot 4.3.stable.official

Here is the code I'm using:

`extends NobodyWhoPromptChat

func _ready() -> void:
	run()
	

func _on_completion_updated(text) -> void:
	print(text)
	$ScrollContainer/Label.text += text

func _on_completion_finished() -> void:
	print("MODEL FINISHED")
	$TextEdit.text = ""
	$Button.visible = true

func _on_button_pressed() -> void:
	print("SAY: ", $TextEdit.text)
	say($TextEdit.text)
	$ScrollContainer/Label.text = ""
	$Button.visible = false
`

**Comment by volesen** [comment]
Hi Jan - which version are you using? The newest version should support longer context length. Also, can you try with Llama 3.2 3B?

**Comment by JanRiebe** [comment]
Probably version 2.0.x loaded from the asset lib. I'll check the latest version asap.

Would be a cool feature if it would throw some error or send a signal, when the context is too large.
Don't know if that's easy to achieve.

**Comment by volesen** [comment]
Definitely! I think we'll release some context summarization very soon as well

**Comment by AsbjornOlling** [comment]
@JanRiebe I expect that this is no longer an issue after the v3.1.0 release.

**Comment by JanRiebe** [comment]
I've just tried with the latest release 3.2.0 and it's still hanging after a while though definitely later than before.
Let me know if I can provide any further info to help.

(btw. I'm not expecting you to fix this, just letting you know that there are issues :)
